extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java

```diff
@@ -234,4 +237,44 @@ private boolean isBinary(String contentType) {
         return false;
     }
 
+    private Optional<Principal> getPrincipal(APIGatewayV2HTTPEvent request) {
+        final Map<String, String> systemEnvironment = System.getenv();
+        final boolean isSamLocal = Boolean.parseBoolean(systemEnvironment.get("AWS_SAM_LOCAL"));
+        final boolean isForceLocalUser = Boolean.parseBoolean(systemEnvironment.get("QUARKUS_AWS_LAMBDA_FORCE_USER"));
+        if (isSamLocal && isForceLocalUser) {
+            final String forcedUserName = systemEnvironment.get("QUARKUS_AWS_LAMBDA_FORCE_USER_NAME");
+            log.info("Forcing local user to " + forcedUserName);
+            return Optional.of(new Principal() {
+
+                @Override
+                public String getName() {
+                    return forcedUserName;
+                }
+
+            });
+        } else {
+            final APIGatewayV2HTTPEvent.RequestContext requestContext = request.getRequestContext();
+            if (requestContext != null) {
+                final APIGatewayV2HTTPEvent.RequestContext.Authorizer authorizer = requestContext.getAuthorizer();
+                if (authorizer != null) {
+                    final APIGatewayV2HTTPEvent.RequestContext.Authorizer.JWT jwt = authorizer.getJwt();
+                    if (jwt != null) {
+                        final Map<String, String> claims = jwt.getClaims();
+                        if (claims != null) {
+                            final String jwtUsername = claims.get("cognito:username");
```

Comment:
Can this `cognito:username` property be configurable ? Maybe we'd like to extract some other info from the claims instead (sub, email,...). Would a config prop, defaulted to `cognito:username`, be cool to have ?

Reply:
Let's do a separate PR for something like that.  This PR is just focused on Principal name.